app-shells/nushell/nushell-0.60.0.ebuild

```diff
@@ -0,0 +1,554 @@
+	streaming-iterator-0.1.5
+	strength_reduce-0.2.3
+	string_cache-0.8.3
+	string_cache_codegen-0.5.1
+	strip-ansi-escapes-0.1.1
+	strum-0.23.0
+	strum-0.24.0
+	strum_macros-0.23.1
+	strum_macros-0.24.0
+	supports-color-1.3.0
+	supports-hyperlinks-1.2.0
+	supports-unicode-1.0.2
+	sxd-document-0.3.2
+	sxd-xpath-0.4.2
+	syn-1.0.86
+	sys-locale-0.1.0
+	sysinfo-0.23.5
+	tempdir-0.3.7
+	tempfile-3.3.0
+	tendril-0.4.2
+	termcolor-1.1.3
+	terminal_size-0.1.17
+	termtree-0.2.4
+	textwrap-0.14.2
+	thin-slice-0.1.1
+	thiserror-1.0.30
+	thiserror-impl-1.0.30
+	time-0.1.44
+	tinyvec-1.5.1
+	tinyvec_macros-0.1.0
+	titlecase-1.1.0
+	tokio-1.17.0
+	tokio-native-tls-0.3.0
+	tokio-util-0.6.9
+	toml-0.5.8
+	tower-service-0.3.1
+	tracing-0.1.32
+	tracing-core-0.1.23
+	trash-2.0.4
+	try-lock-0.2.3
+	typed-arena-1.7.0
+	typenum-1.15.0
+	typetag-0.1.8
+	typetag-impl-0.1.8
+	ucd-trie-0.1.3
+	umask-1.0.1
+	uncased-0.9.6
+	unicode-bidi-0.3.7
+	unicode-linebreak-0.1.2
+	unicode-normalization-0.1.19
+	unicode-segmentation-1.9.0
+	unicode-width-0.1.9
+	unicode-xid-0.2.2
+	url-2.2.2
+	users-0.11.0
+	utf-8-0.7.6
+	utf8-width-0.1.5
+	utf8parse-0.2.0
+	uuid-0.8.2
+	vcpkg-0.2.15
+	version_check-0.1.5
+	version_check-0.9.4
+	void-1.0.2
+	vswhom-0.1.0
+	vswhom-sys-0.1.1
+	vte-0.10.1
+	vte_generate_state_changes-0.1.1
+	wait-timeout-0.2.0
+	walkdir-2.3.2
+	want-0.3.0
+	wasi-0.9.0+wasi-snapshot-preview1
+	wasi-0.10.0+wasi-snapshot-preview1
+	wasm-bindgen-0.2.79
+	wasm-bindgen-backend-0.2.79
+	wasm-bindgen-futures-0.4.29
+	wasm-bindgen-macro-0.2.79
+	wasm-bindgen-macro-support-0.2.79
+	wasm-bindgen-shared-0.2.79
+	web-sys-0.3.56
+	which-4.2.4
+	winapi-0.3.9
+	winapi-i686-pc-windows-gnu-0.4.0
+	winapi-util-0.1.5
+	winapi-x86_64-pc-windows-gnu-0.4.0
+	windows-0.9.1
+	windows-sys-0.30.0
+	windows-sys-0.32.0
+	windows_aarch64_msvc-0.30.0
+	windows_aarch64_msvc-0.32.0
+	windows_gen-0.9.1
+	windows_i686_gnu-0.30.0
+	windows_i686_gnu-0.32.0
+	windows_i686_msvc-0.30.0
+	windows_i686_msvc-0.32.0
+	windows_macros-0.9.1
+	windows_x86_64_gnu-0.30.0
+	windows_x86_64_gnu-0.32.0
+	windows_x86_64_msvc-0.30.0
+	windows_x86_64_msvc-0.32.0
+	winreg-0.7.0
+	winreg-0.10.1
+	xmlparser-0.13.3
+	yaml-rust-0.4.5
+	zeroize-1.5.3
+	zip-0.5.13
+	zstd-0.10.0+zstd.1.5.2
+	zstd-safe-4.1.4+zstd.1.5.2
+	zstd-sys-1.6.3+zstd.1.5.2
+"
+
+inherit cargo
+
+DESCRIPTION="A new type of shell, written in Rust"
+HOMEPAGE="https://www.nushell.sh"
+SRC_URI="https://github.com/nushell/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz
+	$(cargo_crate_uris ${CRATES})"
+
+LICENSE="Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD-2 BSD Boost-1.0 CC0-1.0 ISC MIT MPL-2.0 Unlicense ZLIB"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~ppc64 ~riscv"
+IUSE="+extra"
+
+DEPEND="
+	>=dev-libs/libgit2-0.99:=
+	dev-libs/oniguruma:=
+	dev-libs/openssl:0=
+	net-libs/libssh2:=
+	net-libs/nghttp2:=
+	net-misc/curl
+	extra? (
+		dev-db/sqlite:3=
+		x11-libs/libX11
+		x11-libs/libxcb
+	)
+"
+
+RDEPEND="${DEPEND}"
+
+BDEPEND="
+	>=virtual/rust-1.51
```

Comment:
I think I like the idea of waiting for 1.59 to stabilize, so I'll apply this patch for now:

```
diff --git a/app-shells/nushell/nushell-0.60.0.ebuild b/app-shells/nushell/nushell-0.60.0.ebuild
index 72750538a17..a6a1a3db92e 100644
--- a/app-shells/nushell/nushell-0.60.0.ebuild
+++ b/app-shells/nushell/nushell-0.60.0.ebuild
@@ -520,6 +520,14 @@ BDEPEND="
 
 QA_FLAGS_IGNORED="usr/bin/nu.*"
 
+src_prepare() {
+       default
+
+       # This can be dropped once Rust 1.59.0 is stable, but when we drop this we should update the
+       # BDEPEND to require 1.59.0 as well.
+       sed -i '/^strip = .*/d' Cargo.toml
+}
+
 src_configure() {
        # high magic to allow system-libs
        export OPENSSL_NO_VENDOR=true
```

Reply:
On a side-note, this only works because upstream hasn't set the rust_version in Cargo.toml (otherwise would need to change that too, broot did for instance).

Reply:
I opened https://github.com/nushell/nushell/pull/4940 to set the `rust-version` upstream since technically their `Cargo.toml` does require it.